fdb-relational-core/src/main/java/com/apple/foundationdb/relational/recordlayer/query/PlannerConfiguration.java

```diff
@@ -49,47 +49,70 @@
 public final class PlannerConfiguration {
 
     @Nonnull
-    private static final PlannerConfiguration ALL_AVAILABLE_INDEXES = new PlannerConfiguration(Optional.empty());
+    private final Optional<Set<String>> readableIndexes;
 
     @Nonnull
-    private final Optional<Set<String>> readableIndexes;
+    private final RecordQueryPlannerConfiguration recordQueryPlannerConfiguration;
+
+    private final int memoizedHash;
 
-    private PlannerConfiguration(@Nonnull Optional<Set<String>> readableIndexes) {
+    private PlannerConfiguration(@Nonnull final Optional<Set<String>> readableIndexes,
+                                 @Nonnull final RecordQueryPlannerConfiguration recordQueryPlannerConfiguration) {
         this.readableIndexes = readableIndexes;
+        this.recordQueryPlannerConfiguration = recordQueryPlannerConfiguration;
+        this.memoizedHash = computeHash();
     }
 
     @Nonnull
     public Optional<Set<String>> getReadableIndexes() {
         return readableIndexes;
     }
 
+    @Nonnull
+    public RecordQueryPlannerConfiguration getRecordQueryPlannerConfiguration() {
+        return recordQueryPlannerConfiguration;
+    }
+
     @Override
-    public boolean equals(Object o) {
-        if (this == o) {
+    public boolean equals(Object other) {
```

Comment:
Essentially this is the important part of this PR.

`PlannerConfiguration` is added to `QueryCacheKey`.

The implementation of `equals` and _memoized_ `hashCode` picks the bits that are only relevant for the `QueryCacheKey`, they are:

1. The `IndexFetchMethod`.
2. The set of disabled transformation rules.

cc @normen662 @alecgrieser